src/main/webapp/resources/js/pages/projects/samples-metadata-import/components/SampleMetadataImportSelectFile.tsx

```diff
@@ -26,7 +26,10 @@ export function SampleMetadataImportSelectFile(): JSX.Element {
   const dispatch: ImportDispatch = useImportDispatch();
   const [status, setStatus] = React.useState<StepsProps["status"]>("process");
   const [loading, setLoading] = React.useState<boolean>(false);
-  const [valid, setValid] = React.useState<boolean>(true);
+  const [showEmptyHeaderError, setShowEmptyHeaderError] =
```

Comment:
Instead of using these constants lets replace it with a single one such as `const [uploadValidationErrors, setUploadValidationErrors] = React.useState<ReactElement[]>([]);` Then below we would update line 101 and 104 to push the list item with the relevant text into an array  (such as const validationErrors = []). After line 106 we setUploadValidationErrors(validationErrors).

Reply:
Please see a54ec79fd261822b9a8c159c45f7c4db4a1fb00e.